dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java

```diff
@@ -93,16 +94,25 @@ public void join(WorkerInfo workerInfo) throws IOException {
     if (existingEntityBytes != null) {
```

Comment:
https://github.com/Alluxio/alluxio/blob/87702fd02be834dae29078f4f2d4c8a5e558864a/dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java#L94-L95

https://github.com/Alluxio/alluxio/blob/87702fd02be834dae29078f4f2d4c8a5e558864a/dora/core/common/src/main/java/alluxio/membership/EtcdMembershipManager.java#L118-L120

is there a time-of-check to time-of-use race between these lines? Two workers with conflicting IDs may be registering at the same time, and both can happen to see no conflicting records exist on etcd so proceed to register with the ID, accidentally overwriting each other. I assume etcd transaction is not used for registering a worker.

Reply:
yeah u r right, wasn't using txn coz before ur workerid change, no two workers will have same workernetaddr, which was used as the unique identifier then, to begin with. Will change to txn semantics here, thanks

Reply:
@dbw9580  can u double check this logics to handle race for me plz?